src/cargo/ops/cargo_generate_lockfile.rs

```diff
@@ -89,6 +89,21 @@ pub fn update_lockfile(manifest_path: &Path,
                                                   Method::Everything,
                                                   Some(&previous_resolve),
                                                   Some(&to_avoid)));
+    for dep in compare_dependency_graphs(&previous_resolve, &resolve) {
+        try!(match dep {
+            (None, Some(pkg)) => opts.config.shell().status("Adding",
+                format!("{} v{}", pkg.name(), pkg.version())),
+            (Some(pkg), None) => opts.config.shell().status("Removing",
+                format!("{} v{}", pkg.name(), pkg.version())),
+            (Some(pkg1), Some(pkg2)) => {
+                if pkg1.version() != pkg2.version() {
+                    opts.config.shell().status("Updating",
+                        format!("{} v{} -> v{}", pkg1.name(), pkg1.version(), pkg2.version()))
+                } else {Ok(())}
+            }
+            (None, None) => unreachable!(),
+        });
```

Comment:
This may be better structured like:

```rust
let (status, msg) = match dep {
    (None, Some(pkg)) => ("Adding", format!(...)),
    (Some(pkg), None) => ("Removing", format!(...)),
    (Some(pkg1), Some(pkg2)) => {
        if pkg1.version() == pkg2.version() { continue }
        ("Updating", format!(...))
    }
    (None, None) => continue,
};
try!(opts.config.shell().status(...))
```